src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-cluster-logic.h

```diff
@@ -0,0 +1,221 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/**
+ * @file Cross-platform API to handle cluster-specific logic for the valve configuration and control cluster on a single endpoint.
+ */
+
+#pragma once
+
+#include "valve-configuration-and-control-delegate.h"
+#include "valve-configuration-and-control-matter-context.h"
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/cluster-building-blocks/QuieterReporting.h>
+#include <app/data-model/Nullable.h>
+#include <lib/core/CHIPError.h>
+#include <system/SystemLayer.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ValveConfigurationAndControl {
+
+struct ClusterConformance
+{
+    inline bool HasFeature(Feature feature) const { return featureMap & to_underlying(feature); }
+    uint32_t featureMap;
+    bool supportsDefaultOpenLevel;
```

Comment:
feature map and all following bool supports should be a bitfield
maybe something like this
```
union {
    struct {
        bool supportsDefaultOpenLevel: 1;
        bool supportsValveFault: 1;
        bool supportsLevelStep: 1;
    };
    uint32_t featureMap;
};
```
I'm not sure that it is portable tho. Maybe Using bitflag and enum type would be best